email/1.3/modules/ROOT/pages/email-documentation.adoc

```diff
@@ -0,0 +1,948 @@
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use | | x
+| From Address a| String |  The "From" address of the message sender. If not set, it defaults to the "From" address specified in the config. |  |
+| To Addresses a| Array of String |  The recipient addresses of "To" (primary) type |  | x
+| Cc Addresses a| Array of String |  The recipient addresses of "CC" (carbon copy) type |  |
+| Bcc Addresses a| Array of String |  The recipient addresses of "BCC" (blind carbon copy) type |  |
+| Reply To Addresses a| Array of String |  The email addresses to which to reply to this email |  |
+| Subject a| String | The subject of the email |  [No Subject] |
+| Headers a| Object | The headers that this email carries |  |
+| Content a| Binary | Text body of the message, which can be in any format |  `#[payload]` |
+| ContentType a| String |  Content Type of the body text. Example: "text/plain" |  |
+| Encoding a| String |  The character encoding of the body. If it is configured, it overrides the one inferred from the body. |  |
+| Attachments a| Object |  The attachments that are sent along with the email body |  |
+| Content Transfer Encoding a| String |  Encoding used to indicate the type of transformation that is used to represent the body in an acceptable manner for transport. The value is case insensitive.  Known encodings:  BASE64, QUOTED-PRINTABLE, 8BIT, 7BIT, BINARY  |  `Base64` |
+| Reconnection Strategy a| * <<reconnect>>
+* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
+|===
+
+
+=== For Configurations
+* <<smtp>>
+
+==== Throws
+* EMAIL:SEND
+* EMAIL:CONNECTIVITY
+* EMAIL:RETRY_EXHAUSTED
+
+
+== Sources
+
+[[listener-imap]]
+=== On New Email - IMAP
+`<email:listener-imap>`
+
+
+Retrieves all the emails from an IMAP mailbox folder, watermark can be enabled for polled items.
+
+
+==== Parameters
+[%header,cols="20s,20a,35a,20a,5a"]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use. | | x
+| Folder a| String |  The name of the folder to poll emails from.  |  `INBOX` |
+| Delete After Retrieve a| Boolean |  Enables the deletion of the polled emails after being retrieved. This is disabled by default. | `false` |
+| Enable Watermark a| Boolean |  If watermark should be applied to the polled emails or not. Default to true. |  true |
+| Imap Matcher a| <<imap-matcher>> |  A matcher to filter emails retrieved by this polling source. By default already read emails will be filtered. |  |
+| Enable Remote Search a| Boolean |  If search filters should be resolved on the server side the value is `true`, or on the client side the value is `false`. Default to `false` because some email servers are not fully compliant with `rfc-3501` search terms. Activating this feature diminishes traffic by reducing the amount of emails brought to the client side for processing. |  `false` |
```

Comment:
```suggestion
| Enable Remote Search a| Boolean |  If search filters should be resolved on the server side, set this value to `true`. The default is `false`, which means search filters are resolved on the client side. This is because some email servers are not fully compliant with `rfc-3501` search terms. Activating this feature diminishes traffic by reducing the number of emails sent to the client side for processing. |  `false` |
```